backend/server/routers/specialisations.py

```diff
@@ -0,0 +1,81 @@
+from fastapi import APIRouter, HTTPException
+from server.database import programsCOL
+from server.routers.model import SpecialisationTypes, Specialisations
+
+router = APIRouter(
+    prefix="/specialisations",
+    tags=["specialisations"],
+)
+
+
+@router.get("/")
+def specialisationsIndex():
+    """ sanity test that this file is loaded """
+    return "Index of specialisations"
+
+@router.get(
+    "/getSpecialisationTypes/{programCode}",
+    response_model=SpecialisationTypes,
+    responses={
+        200: {"types": ["majors", "minors"]}
+    }
+)
+def get_specialisation_types(programCode):
+    result = programsCOL.find_one({"code": programCode})
+
+    if not result:
+        raise HTTPException(
+            status_code=400, detail="Program code was not found")
+    return {"types": [*result["components"]["spec_data"].keys()]}
+
+
+@router.get(
+    "/getSpecialisations/{programCode}/{typeSpec}",
+    response_model=Specialisations,
+    responses={
+        400: {
+            "description": "The given program code could not be found in the database",
+        },
+        200: {
+            "description": "Returns all majors to the given code",
+            "content": {
+                "application/json": {
+                    "example": {
+                        "spec": {
```

Comment:
shouldn’t it be `majors` here?

Reply:
nope - ive unified this so it can be a major, minor, or honours depending on request.